e2e/claim_test.go

```diff
@@ -82,15 +84,57 @@ func TestClaim(t *testing.T) {
 	require.NotNil(t, res)
 	require.Equal(t, res.CodeID, uint64(3))
 
+	// marketplace
+	resp, err := grab.Get("./contracts/", "https://github.com/public-awesome/marketplace/releases/download/v0.5.0/sg_marketplace.wasm")
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println("Download saved to", resp.Filename)
+	b, err = resp.Bytes()
+	require.NoError(t, err)
+
+	res, err = msgServer.StoreCode(sdk.WrapSDKContext(ctx), &wasmtypes.MsgStoreCode{
+		Sender:       addr1.String(),
+		WASMByteCode: b,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, res)
+	require.Equal(t, res.CodeID, uint64(4))
+
 	creator := accs[0]
 
+	instantiateMsgRaw := []byte(
+		fmt.Sprintf(instantiateMarketplaceTemplate,
+			2,
+			86400,
+			15552000,
+			86400,
+			15552000,
+			creator.Address.String(),
+		),
+	)
+
+	// instantiate marketplace
+	instantiateRes, err := msgServer.InstantiateContract(sdk.WrapSDKContext(ctx), &wasmtypes.MsgInstantiateContract{
+		Sender: creator.Address.String(),
+		Admin:  creator.Address.String(),
+		CodeID: 4,
+		Label:  "Marketplace",
+		Msg:    instantiateMsgRaw,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, instantiateRes)
+	require.NotEmpty(t, instantiateRes.Address)
+	marketplaceAddress := instantiateRes.Address
+	require.NotEmpty(t, marketplaceAddress)
+
```

Comment:
I guess the e2e testing for buying an nft would be pending? we can track it on a different issue

Reply:
@jhernandezb https://github.com/public-awesome/marketplace/issues/91